archive_feedstock.py

```diff
@@ -28,57 +38,108 @@ def archive_repo(owner, repo, archive=True, check_only=False):
     )
     raise_json_for_status(r)
 
-    target_status = f"{'' if archive else 'un'}archived"
+    if task == "archive":
+        target_status = "archived"
+    else:
+        target_status = "unarchived"
+
     data = r.json()
-    if data["archived"] == archive:
-        raise RuntimeError(f"{owner}/{repo} is already {target_status}!")
-    elif check_only:
+    if task == "archive" and data["archived"]:
+        print("feedstock %s is already %s" % (repo, target_status), flush=True)
+        return
+
+    if task == "unarchive" and not data["archived"]:
+        print("feedstock %s is already %s" % (repo, target_status), flush=True)
         return
 
     r = requests.patch(
         f"https://api.github.com/repos/{owner}/{repo}",
         headers=headers,
-        json={"archived": archive}
+        json={"archived": True if task == "archive" else False}
     )
     raise_json_for_status(r)
 
-
-def feedstocks(directory="archive"):
-    for path in Path(directory).glob("*.txt"):
-        if path.name == "example.txt":
-            continue
-        with open(path) as f:
-            for line in f:
-                line = line.strip()
-                if line and not line.startswith("#"):
-                    yield line
+    print("feedstock %s was %s" % (repo, target_status), flush=True)
 
 
-def main(owner="conda-forge", check_only=False):
-    exceptions = []
-    for task in "archive", "unarchive":
-        seen = set()
-        for feedstock in feedstocks(task):
-            verb = "Archiving" if task == "archive" else "Unarchiving"
-            print(f"{verb} {owner}/{feedstock}-feedstock...", end=" ")
-            if feedstock in seen:
-                print("[SKIP]")
+def process_feedstocks_in_file(task_file, task):
+    pkgs_to_do_again = []
+    with open(task_file, "r") as fp:
+        for line in fp.readlines():
```

Comment:
```suggestion
        for line in fp:
```

Wouldn't this do the same?